pkg/datasources/account_roles.go

```diff
@@ -0,0 +1,94 @@
+package datasources
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var accountRolesSchema = map[string]*schema.Schema{
+	"pattern": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Filters the command output by object name.",
+	},
+	"roles": {
```

Comment:
another idea i had was instead of calling this roles, calling it `result` or `rows`.So instead of `data.snowflake_account_roles.roles.roles` we would have `data.snowflake_account_roles.roles.result`.

Reply:
Same as below